model/src/data-model.js

```diff
@@ -157,6 +157,32 @@ class Data {
     return this
   }
 
+  addComponent (pagePath, component) {
+    const page = this.findPage(pagePath)
+    if (page) {
+      page.components = page.components || []
+      page.components.push(component)
+    } else {
+      throw Error(`No page exists with path ${pagePath}`)
```

Comment:
are you doing anything with the thrown errors? or just printing them?

Reply:
I think they're just printing at the moment. I guess it's down to us what our strategy should be in each system (designer / runner) but it feels like the model should throw and let the caller decide what to do in that situation